isis/scripts/install_isis.sh

```diff
@@ -0,0 +1,355 @@
+#!/bin/bash
+
+# Define environment name and package to install
+ISIS_LABEL="main"
+DOWNLOAD_DATA="YES"
+
+# Report error and reserves error code from a failed command
+failed_command() {
+    local _status=$?
+    if [ "${_status}" -ne 0 ]; then 
+        echo "Error: $1 failed" 
+        exit ${_status}
+    fi
+}
+
+print_help() {
+    printf "Usage: $0 [options] ISIS_LABEL ISIS_VERSION\n"
+    printf "Options:\n"
+    printf "\t-h, --help            Show this help message and exit\n"
+    printf "\t-n, --env_name        The name of the anaconda environment to create.\n"
+    printf "\t-m, --miniforge_dir   Define the directory to an anaconda package manager install location. "
+    printf "If you have an anaconda package manager already this argument will be ignored. If not "
+    printf "a version of miniforge will be installed at this location\n"
+    printf "\t-p, --data_prefix     The directory where ISISDATA is located. If this "
+    printf "directory doesn't exist then one will be made at its location\n"
+    printf "\t--no_data             Do not ask to download any data for the ISIS_DATA area.\n"
+    printf "\n"
+    printf "\tDefining variables on the command line will skip the interactive "
+    printf "elements within this script"
+    printf "\n\n"
+    printf "\t\$1 ISIS_LABEL - Different ISIS labels as defined by the "
+    printf "anaconda labels at https://anaconda.org/usgs-astrogeology/isis"
+    printf ", examples include \"LTS\", \"dev\", and \"RC\"\n"
+    printf "\t\$2 ISIS_VERSION - Different ISIS versions as defined by the "
+    printf "anaconda versions for a label at https://anaconda.org/usgs-astrogeology/isis"
+    printf ", examples include 8.0.3 (LTS/Feature), 8.2.0_RC1 (RC), and 2025.02.22 (dev)"
+    exit 1
+}
+
+POSITIONAL_ARGS=()
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -h|--help)
+      print_help
+      ;;
+    -n|--env_name)
+      ENV_NAME="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    -m|--miniforge_dir)
+      MINIFORGE_DIR="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    -p|--data_prefix)
+      ISISDATA_PREFIX="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --no_data)
+      DOWNLOAD_DATA=NO
+      shift # past argument
+      ;;
+    -*|--*)
+      echo "Unknown option $1"
+      exit 1
+      ;;
+    *)
+      POSITIONAL_ARGS+=("$1") # save positional arg
+      shift # past argument
+      ;;
+  esac
+done
+
+set -- "${POSITIONAL_ARGS[@]}" # restore positional parameters
+
+# If a ISIS LABEL was given, set it
```

Comment:
`ISIS_LABEL` and `ISIS_VERSION` are not being handled as intended. If a user only intends to specify an ISIS version and therefore only passes a single positional parameter, the script will set that parameter as the value of `ISIS_LABEL`, not `ISIS_VERSION`.

By convention, positional parameters are required. In this script, `ISIS_LABEL` and `ISIS_VERSION` are meant to be optional, and thus should be added as command line options that each take a single argument.

Also: consider naming `ISIS_LABEL` something else. When I see "ISIS label," I think of detached labels associated with an ISIS cube file, not the label on a package in conda.